core/networkserver/manager_server.go

```diff
@@ -23,8 +23,8 @@ type networkServerManager struct {
 }
 
 func (n *networkServerManager) getDevice(ctx context.Context, in *pb_lorawan.DeviceIdentifier) (*device.Device, error) {
-	if !in.Validate() {
-		return nil, errors.NewErrInvalidArgument("Device Identifier", "validation failed")
+	if err := in.Validate(); err != nil {
+		return nil, errors.BuildGRPCError(errors.NewErrInvalidArgument("Device Identifier", err.Error()))
```

Comment:
Be consistent with `errors.Wrap` vs wrapping it in `errors.NewErrInvalidArgument`